articles/qos_deadline_liveliness_lifespan.md

```diff
@@ -0,0 +1,235 @@
+To address these needs it was proposed that we start by adding support for the following new DDS QoS settings.
+
+### Deadline
+
+The deadline policy establishes a contract for the amount of time allowed between messages.
+For Subscriptions it establishes the maximum amount of time allowed to pass between receiving messages.
+For Publishers it establishes the maximum amount of time allowed to pass between sending messages.
+For Service Servers it establishes the maximum amount of time allowed to pass between receiving a request and when a response for that request is sent.
+For Service Clients it establishes the maximum amount of time allowed to pass between sending a request and when a response for that request is received.
+
+Topics and Services will support the following levels of deadlines:
+- DEADLINE_DEFAULT - Use the ROS specified default for deadlines (which is DEADLINE_NONE).
+- DEADLINE_NONE - No deadline will be offered or requested and events will not be generated for missed deadlines.
+- DEADLINE_RMW - The rmw layer of ROS will track the deadline. For a Publisher or Subscriber this means that a deadline will be considered missed if the rmw layer has not received a message within the specified time. For a Service this means the time a request is started is marked when the request reaches the rmw layer and the time at which it is finished is when the response message reaches the rmw layer.
+
+In order for a Subscriber to listen to a Publisher's Topic the deadline that they request must be greater than, or equal to, the deadline set by the Publisher.
+A Service Client will **not** be prevented from making a request to a Service Server if the  Server provides a deadline greater than the deadline requested by the Client.
+
+### Liveliness
+
+The liveliness policy establishes a contract for how entities report that they are still alive. 
+For Subscriptions it establishes the level of reporting that they require from the Publishers to which they are subscribed. 
+For Publishers it establishes the level of reporting that they will provide to Subscribers that they are alive. 
+For Service Servers it establishes both the level of reporting that they will provide to Clients and also the level of reporting that they require from Clients.
+For Service Clients it establishes both the level of reporting that they require from Service Servers and the level of reporting that they will provide to the Server.
+
+Topics will support the following levels of liveliness:
+- LIVELINESS_DEFAULT - Use the ROS specified default for liveliness (which is LIVELINESS_AUTOMATIC).
+- LIVELINESS_AUTOMATIC - The signal that establishes a Topic is alive comes from the ROS rmw layer.
+- LIVELINESS_MANUAL_NODE - The signal that establishes a Topic is alive is at the node level. Publishing a message on any outgoing channel on the node or an explicit signal from the application to assert liveliness on the node will mark all outgoing channels on the node as being alive.
```

Comment:
Note that one should decouple participant and nodes for performance and resource reasons in the near future. This will then complicate the implementation of this now seemingly trivial feature.